### PR TITLE
fix(dapp): makes sig validation compatible with ERC-1271

### LINKED ISSUE #117: [dApp] Problem with smart contract wallet integration
We have an NFT renting product, [rntd.io](https://rntd.io/) ,This product is still under development. We are still in the alpha version, we used walletConnect V1 to integrate our smart contract wallet with projects.
In V1, we did not face any problems with the integration with other projects, we used to authenticate personal_sign and typed_data to our smart contract with no effort due to verifySignature function that checks if the address is EOA address or smart contract address
https://github.com/WalletConnect/walletconnect-example-dapp/blob/master/src/helpers/utilities.ts#L181

But in V2, the provided dApp examples use  verifyEip155MessageSignature, which is only valid for EOA wallets, not for smart contract wallets that rely on the concept of erc1271.
https://github.com/WalletConnect/web-examples/blob/main/dapps/react-dapp-v2-with-ethers/src/pages/index.tsx#L69
When the function recovers the address of the wallet that signed the message, which owns the smart contract wallet, and compare it to the smart contract address, the validation will fail. As a result, we can not verify the personal message.
My question is, what are we missing, or what can we do to overcome this problem

### PR DESCRIPTION
**Pull request by bkrem**
- Fixes #117 
- Supersedes #140 
- Tested that existing signature validation still works as expected with https://react-wallet.walletconnect.com, on top of the additional ERC/EIP 1271 validation now accounted for.

### CODE REVIEW

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx [hunk 1]
```diff
@@ -315,10 +317,14 @@ export function JsonRpcContextProvider({
           throw new Error(`Missing chain data for chainId: ${chainId}`);
         }
 
-        const valid = _verifyEip155MessageSignature(
-          message,
+        const hashMsg = hashPersonalMessage(message);
+        const rpc =
```
Comment (Cali93) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:321]: Should we throw an error if there's no rpcProvider for the given chainId ?

**Review by Cali93 — APPROVED** [review]
LGTM, just a minor question around RPC url.

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx [hunk 2]
```diff
@@ -315,10 +317,14 @@ export function JsonRpcContextProvider({
...
+          rpcProvidersByChainId[Number(chainId.toString().split(":")[1])];
```
Comment (Cali93) [inline dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:322]: Should we throw an error if there's no rpcProvider for the given chainId ?

**Review by Cali93 — APPROVED** [review]
LGTM, just a minor question around RPC

File: dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx (hunk 2 above)
Comment (bkrem) [reply dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx]: Good call thank you. Actually helped me clear up the existing check there which was redundant 👍

**Pull request merged**